python/ray/util/tracing/tracing_helper.py

```diff
@@ -328,7 +328,9 @@ def _inject_tracing_into_function(function):
     future execution of that function will include tracing.
     Use the provided trace context from kwargs.
     """
-    # Add _ray_trace_ctx to function signature
+    if not _is_tracing_enabled():
```

Comment:
This change means that the change up there in remote_function.py is just for the optimization when tracing is enabled right? Just with this fix, if the tracing is not enabled, there would be no regression

Reply:
Just with this, there would be no regression, but there would be a correctness issue because `_is_tracing_enabled()` is not accurate at the time this function was originally being called in `remote_function.py` (needs to be after `init()`)